### PR TITLE
getContent endpoint

### PR DESCRIPTION
**Pull request by MarwanGalal746**
I added: 
/src/web/main.g
/src/pkg/handlers/handlers.go
/src/pkg/content/content.go
I downloaded an external library named gorilla for routing from here: https://github.com/gorilla/mux

### CODE REVIEW

**Comment by MarwanGalal746** [comment]
getContent endpoint with DB has been added

**Comment by odetolakehinde** [comment]
@MarwanGalal746 can you please fix this conflicts?

And attach the issue to this PR.

File: migrations/20210817115335_create_db_inits.postgres.up.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE contents (
+   id bigserial PRIMARY KEY,
+   title varchar(150) NOT NULL,
+   details TEXT NOT NULL,
+   created_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
+   updated_at TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP
```
Comment (odetolakehinde) [inline migrations/20210817115335_create_db_inits.postgres.up.sql:6]: `updated_at` should have default as null.

File: migrations/schema.sql [hunk 2]
```diff
@@ -0,0 +1,95 @@
+--
```
Comment (odetolakehinde) [inline migrations/schema.sql:1]: Let's try to use this package to manage our migrations. It makes life easier.

Here is the link: https://github.com/pressly/goose

File: src/pkg/config/config.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package config
+
+import (
+	"database/sql"
+	"fmt"
+	_ "github.com/joho/godotenv"
+	_ "github.com/lib/pq"
+	"os"
+	"strconv"
+	_ "strconv"
```
Comment (odetolakehinde) [inline src/pkg/config/config.go:10]: why import this if you are not using it?

File: src/pkg/handlers/handlers.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	}
+	var res []content.Content
+	if item.ID != "" {
+		res = append(res, item)
+	}
+	return res
+}
+
+//utility function to get a specific range of contents
+func getRangeOfContents(ids []string) []content.Content {
```
Comment (odetolakehinde) [inline src/pkg/handlers/handlers.go:93]: this should take two ids instead of a slice.

File: src/pkg/handlers/handlers.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	for i := 0; i < len(items); i++ {
+		json.NewEncoder(w).Encode(items[i])
+	}
+}
+
+//utility function to get a specific content frim db
+func getOneContent(id string) []content.Content {
```
Comment (odetolakehinde) [inline src/pkg/handlers/handlers.go:72]: this should return `content.Content` instead of a slice of content. Since it will always return just one.

**Review by odetolakehinde — CHANGES_REQUESTED** [review]
In general, I think the controllers or handlers should be separated from the database functions so you can use it better. Also, consider using an ORM such as https://gorm.io/index.html to avoid writing all these complex queries.

File: migrations/20210817115335_create_db_inits.postgres.up.sql (hunk 1 above)
Comment (MarwanGalal746) [reply migrations/20210817115335_create_db_inits.postgres.up.sql:6]: This is old PR, please close it and don't merge it

**Pull request closed without merging**